test_crates/enum_now_doc_hidden/new/src/lib.rs

```diff
@@ -0,0 +1,83 @@
+mod MyNonPublicMod {
+    // despite adding #[doc(hidden)], this enum is in a
+    // private mod, so it isn't part of the crate's public
+    // api
+    #[doc(hidden)]
+    pub enum MyEnum {
+        A,
+    }
+}
+
+pub mod MyPublicMod {
+    // added #[doc(hidden)], however this enum is in a
+    // public mod, so it is part of the crate's public api
+    #[doc(hidden)]
+    pub enum MyEnum {
+        A,
+    }
+}
+
+mod MyNestedNonPublicMod {
+    pub mod PublicInnerStruct {
+        // despite adding #[doc(hidden)], this enum is in a
+        // private outer mod, so it isn't part of the crate's public
+        // api
```

Comment:
As a variation of this test, you can try using a `pub mod` that is itself `#[doc(hidden)]` and then adding `#[doc(hidden)]` on the enum held inside. That shouldn't get flagged because the enum wasn't public API originally — you would have to had imported it from a hidden module.